main.js

```diff
@@ -0,0 +1,174 @@
+let dogEndpoint = 'https://dog.ceo/api/breeds/image/random';
+let history = [];
+let h_input = document.getElementById('input');
+let h_output = document.getElementById('output');
+
+//object containing inputs and outputs arrays
+let chatInputsOutputs = [
+  {
+    inputs: ['Hello', 'Hi', 'Greetings'],
+    outputs: ['Hello', 'Hey', 'Greetings'],
+  },
+  {
+    inputs: [
+      'What is your favourite colour?',
+      'Who is your favourite HYF instructor?',
+      'Who is your role model?',
+    ],
+    outputs: [
+      'I am not sure.',
+      'There are too many to choose from.',
+      'I like everyone.',
+    ],
+  },
+  {
+    inputs: [
+      'How are you?',
+      'How is the weather today?',
+      'How is Canada doing in the Olympics?',
+    ],
+    outputs: ['Fine', 'Great', 'Not so good'],
+  },
+];
+
+const randomNuberGenerator = () => Math.floor(Math.random() * 3);
+
+console.log(chatInputsOutputs);
+
+//answer based on the option selected in the radio button
+const answerRandom = item => {
+  return item[0].outputs[randomNuberGenerator()];
+};
+
+const answerShortest = item => {
+  return item[0].outputs.reduce((a, b) => (a.length < b.length ? a : b));
+};
+
+const answerLongest = item => {
+  return item[0].outputs.reduce((a, b) => (a.length > b.length ? a : b));
+};
+
+// reply function
+const reply = selectedAnswer => {
+  let question = h_input.value;
+  let dogLink = '';
+  console.log(question);
+
+  history.push('You: ' + question + '\n');
+
+  console.log(history);
+
+  const filteredObject = chatInputsOutputs.filter(item => {
+    return item.inputs
+      .map(item => item.toLowerCase())
+      .includes(question.toLowerCase());
+  });
+
+  if (question.toLowerCase().includes('dog')) {
+    console.log('if dog question');
+
+    const assignDogLink = () => {
+      return getPhotos();
+    };
+
+    function getPhoto(aa, assignDogLink) {
+      dogLink = assignDogLink();
+    }
+
+    getPhoto(assignDogLink, function() {
```

Comment:
I think you have duplication of logic here, why don't you just use getPhotos directly and assign to a variable?